src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -232,10 +232,14 @@ class OidcAuthorizeService implements AuthorizeService {
         }
     }
 
-    private async stateExists(url: string) {
-        const stateParam = new URLSearchParams(new URL(url).search).get('state');
-        if (stateParam && this._userManager.settings.stateStore) {
-            return await this._userManager.settings.stateStore.get(stateParam);
+    private async stateExists(urlString: string) {
+        const url = new URL(urlString);
+        const responseMode = this._userManager.settings.response_mode;
+        const urlSearchParams = (responseMode === "query" ) ? (new URLSearchParams(url.search)) : (new URLSearchParams(url.hash));
```

Comment:
Can `response_mode` ever be null? (e.g when it's not explicitly set because it's inferred from the response type)

Reply:
```suggestion
        const urlSearchParams = new URLSearchParams(responseMode === "query" ? url.search : url.hash);
```

Reply:
This should be computed based on the response mode (if explicitly set) or the response type otherwise